### PR TITLE
[columnar] HYD-761 Set maximum column size

### PR DESCRIPTION
**Pull request by mkaruza**
Metadata table is on heap. If we try to construct metadata row bigger than 1GB insert will fail at that point. Limit maximum column size to be 256MB.

### CODE REVIEW

File: columnar/src/backend/columnar/columnar_writer.c
Comment (JerrySievert) [inline columnar/src/backend/columnar/columnar_writer.c:288]: we already have customers that are over the 256mb limit, how do we deal with that?

Comment (wuputah) [reply columnar/src/backend/columnar/columnar_writer.c:288]: not sure if we do or not, but to what degree would their tables already broken?

Comment (mkaruza) [reply columnar/src/backend/columnar/columnar_writer.c:288]: It is limit per row/column. Just from previous PR github#221 it looks that they there could be some issues in database. For what is worth, this limit is not related to columnar storage. Same issue can be observed in heap itself

postgres=# create table t_heap(a text, b text);
CREATE TABLE
postgres=# insert into t_heap SELECT repeat('a', 1000000000), repeat('a', 1000000000);
ERROR:  invalid memory alloc request size 2000000056
postgres=# 

So, this is only related when we insert `columnar.chunk`. Complete row in that table should be less than 1GB in size.

**Review by JerrySievert — APPROVED** [review]
code changes a-ok, but we should wait on @wuputah's agreement before merge.

**Review by wuputah — APPROVED** [review]
I'm OK with it.

**Pull request merged**
